### PR TITLE
Implemented Admin async TruncateTable

### PR DESCRIPTION
**Pull request by rupeshitpatekarq**
This Pull Request is created for Truncate Async Table operations and also implemented earlier review comments.

### CODE REVIEW

File: bigtable-hbase-2.x-parent/bigtable-hbase-2.x-integration-tests/src/test/java/com/google/cloud/bigtable/hbase/async/TestAsyncTruncateTable.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+	private AsyncAdmin getAsyncAdmin() throws InterruptedException, ExecutionException {
+    return AbstractAsyncTest.getAsyncConnection().getAdmin();
+  }
+	
+	/**
+   * @throws Exception 
+   */
+  @Test
+  public void testTruncateTableAsync() throws Exception{
+  	TableName newTestTableName = sharedTestEnv.newTestTableName();
```
Comment (sduskis) [inline bigtable-hbase-2.x-parent/bigtable-hbase-2.x-integration-tests/src/test/java/com/google/cloud/bigtable/hbase/async/TestAsyncTruncateTable.java]: You shouldn't need this method.  Please add a `doTruncate()` method in AbstractTestTruncateTable, and override that here.

File: bigtable-hbase-2.x-parent/bigtable-hbase-2.x/src/main/java/com/google/cloud/bigtable/hbase2_x/BigtableAsyncAdmin.java [hunk 2]
```diff
@@ -307,9 +325,18 @@ public BigtableAsyncAdmin(BigtableAsyncConnection asyncConnection) throws IOExce
...
 
+  /* (non-Javadoc)
+   * @see org.apache.hadoop.hbase.client.AsyncAdmin#truncateTable(org.apache.hadoop.hbase.TableName, boolean)
+   */
   @Override
-  public CompletableFuture<Void> truncateTable(TableName arg0, boolean arg1) {
-    throw new UnsupportedOperationException("truncateTable"); // TODO
+  public CompletableFuture<Void> truncateTable(TableName tableName, boolean preserveSplits) {
+  	return CompletableFuture.supplyAsync(() -> {
+      DropRowRangeRequest.Builder dropBuilder = DropRowRangeRequest.newBuilder().setDeleteAllDataFromTable(true);
```
Comment (sduskis) [inline bigtable-hbase-2.x-parent/bigtable-hbase-2.x/src/main/java/com/google/cloud/bigtable/hbase2_x/BigtableAsyncAdmin.java]: I think that you can just create a `DropRowRangeRequest` outside of `CompletableFuture`, and then just return:

```
FutureUtils.toCompletableFuture(bigtableTableAdminClient.dropRowRangeAsync(request))
.thenApply(r -> null);
```

File: bigtable-hbase-parent/bigtable-hbase-integration-tests-common/pom.xml [hunk 3]
```diff
@@ -33,7 +33,7 @@ limitations under the License.
         <dependency>
             <groupId>org.apache.hbase</groupId>
             <artifactId>hbase-client</artifactId>
-            <version>${hbase.version}</version>
+            <version>${hbase.version.2}</version>
```
Comment (sduskis) [inline bigtable-hbase-parent/bigtable-hbase-integration-tests-common/pom.xml]: We can't do this.  This has to stay `hbase.version`.  You need to make all changes related to hbase 2 in the integration-tests project.

File: bigtable-hbase-parent/bigtable-hbase-integration-tests-common/src/main/java/com/google/cloud/bigtable/hbase/AbstractTestAsyncTruncateTable.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+import org.apache.hadoop.hbase.TableName;
+import org.apache.hadoop.hbase.client.Admin;
+import org.junit.Test;
+
+public class AbstractTestAsyncTruncateTable extends AbstractTestTruncateTable {
+
+	@Override
+	protected void createTable(TableName tableName, byte[][] splits)
+			throws IOException {
+		try(Admin admin = getConnection().getAdmin()) {
```
Comment (sduskis) [inline bigtable-hbase-parent/bigtable-hbase-integration-tests-common/src/main/java/com/google/cloud/bigtable/hbase/AbstractTestAsyncTruncateTable.java]: The spacing seems wrong again.

**Comment by rupeshitpatekarq** [comment]
- Instead of creating sub class of AbstractTestTruncateTable, created abstract doTruncate() method in this class and override that in TestAsyncTruncateTable, TestTruncateTable 1.x & 2.x.
- Kept DropRowRangeRequest outside of CompletableFuture
- Corrected spacing issue

**Pull request merged**